swan-lake/resources/featured-scenarios/write-a-restful-api-with-ballerina.md

```diff
@@ -28,76 +26,134 @@ This guide will walk you through creating a RESTful API with two endpoints. In t
+An in-memory dataset with three entries is used to keep things simple. Follow the steps below to add the definition of the record and the declaration of the [table](/learn/by-example/table/) that holds the data.
+
+1. Generate the record type corresponding to the request payload of the REST service by providing `CovidEntry` as the name and the sample JSON payload below.
+
+    >**Note:** You need to update the generated record by adding the pipe symbol to mark the record as a [closed one](https://ballerina.io/learn/by-example/controlling-openness/) and adding the `readonly` descriptor to the `iso_code` field to make it non-modifiable in order to use it as the key of the table.
+
+    ```json
+    {
+        "iso_code": "AFG",
+        "country": "Afghanistan",
+        "cases": 159303,
+        "deaths": 7386,
+        "recovered": 146084,
+        "active": 5833
+    }
+    ```
+
+    ![Create data record](/learn/images/featured-scenarios/write-a-restful-api-with-ballerina/create-data-record.gif)
+
+    The generated record type will be as follows.
+
+    ```ballerina
+    type CovidEntry record {|
+    readonly string iso_code;
+    string country;
+    int cases;
+    int deaths;
+    int recovered;
+    int active;
+    |};
+    ```
+
+2. Create the table, as shown below.
+
+    >**Note:** Enter `CovidEntry` as the row type corresponding to the type of the members of the table, `iso_code` as the key, and `covidTable` as the variable name when creating the table.
+
+    ![Create data table](/learn/images/featured-scenarios/write-a-restful-api-with-ballerina/create-data-table.gif)
+
+3. Replace the `{key: value}` placeholder of the generated table with the following entries to add the initial data to the table.
+
+    ```
+    {iso_code: "AFG", country: "Afghanistan", cases: 159303, deaths: 7386, recovered: 146084, active: 5833},
+    {iso_code: "SL", country: "Sri Lanka", cases: 598536, deaths: 15243, recovered: 568637, active: 14656},
+    {iso_code: "US", country: "USA", cases: 69808350, deaths: 880976, recovered: 43892277, active: 25035097}
+    ```
+
+The generated record and the table will be as follows.
 
 ```ballerina
-public type CovidEntry record {|
+type CovidEntry record {|
     readonly string iso_code;
     string country;
-    decimal cases;
-    decimal deaths;
-    decimal recovered;
-    decimal active;
+    int cases;
+    int deaths;
+    int recovered;
+    int active;
 |};
 
-public final table<CovidEntry> key(iso_code) covidTable = table [
+final table<CovidEntry> key(iso_code) covidTable = table [
     {iso_code: "AFG", country: "Afghanistan", cases: 159303, deaths: 7386, recovered: 146084, active: 5833},
     {iso_code: "SL", country: "Sri Lanka", cases: 598536, deaths: 15243, recovered: 568637, active: 14656},
     {iso_code: "US", country: "USA", cases: 69808350, deaths: 880976, recovered: 43892277, active: 25035097}
-];
+    ];
```

Comment:
There should not be a indentation here right? Ballerina vscode formatter removes this

Reply:
Fixed.